students/dave_arasim/session07/html_render.py

```diff
@@ -0,0 +1,166 @@
+#!/usr/bin/env python
+
+"""
+A class-based system for rendering html.
+"""
+
+############################################################
+# This is the framework for the base class
+############################################################
+
+class Element(object):
+    tag = 'html'
+    doctype = ''
+    indent_level = 0
+    indent_str = '   '
+
+    def __init__(self, content=None, **kwargs):
+        if content is None:
+            self.contents = []
+        else:
+            self.contents = [content]
+
+        self.attrs = kwargs
+
+    def append(self, new_content):
+        self.contents.append(new_content)
+
+    def _open_tag(self):
+        nv_pairs = ''
+
+        if self.attrs is not None:
+            for key,value in self.attrs.items():
+                nv_pairs += ' ' + str(key) + '='
+                nv_pairs += '"' + str(value) + '"'
```

Comment:
Minor: The whole lines 31-34 could be reduced to just 1 line as follows (using list comprehension):
```
nv_pairs = ''.join(' {}={}'.format(key, value) for key, value in self.attrs.items())
```